Non-deterministic failure of test_kraken_multi teardown

Sometimes Travis fails, sometimes it succeeds, on this particular part of the test suite. Probably some race condition as xdist parallelizes the tests.

Mysteries include:
 - Why is the existence of `libgkl_compression2251932993111688424.so` non-deterministic?
 - Why is there a weird number in that filename?
 - Why is `os.unlink` trying to use a compression library to delete files?
 - Will this problem go away by itself? If so, will that make us feel better or worse about it?

```
_______ ERROR at teardown of test_kraken_multi[TestMetagenomicsViralMix] _______
[gw1] linux -- Python 3.6.5 /home/travis/build/broadinstitute/viral-ngs/tools/conda-tools/default/bin/python
def reset():
>       shutil.rmtree(new_tempdir)
conftest.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tools/conda-tools/default/lib/python3.6/shutil.py:480: in rmtree
    _rmtree_safe_fd(fd, path, onerror)
tools/conda-tools/default/lib/python3.6/shutil.py:438: in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
topfd = 11, path = '/tmp/pytest-of-travis/pytest-0/popen-gw1/test-function0'
onerror = <function rmtree.<locals>.onerror at 0x7f004cc5f1e0>
    def _rmtree_safe_fd(topfd, path, onerror):
        names = []
        try:
            names = os.listdir(topfd)
        except OSError as err:
            err.filename = path
            onerror(os.listdir, path, sys.exc_info())
        for name in names:
            fullname = os.path.join(path, name)
            try:
                orig_st = os.stat(name, dir_fd=topfd, follow_symlinks=False)
                mode = orig_st.st_mode
            except OSError:
                mode = 0
            if stat.S_ISDIR(mode):
                try:
                    dirfd = os.open(name, os.O_RDONLY, dir_fd=topfd)
                except OSError:
                    onerror(os.open, fullname, sys.exc_info())
                else:
                    try:
                        if os.path.samestat(orig_st, os.fstat(dirfd)):
                            _rmtree_safe_fd(dirfd, fullname, onerror)
                            try:
                                os.rmdir(name, dir_fd=topfd)
                            except OSError:
                                onerror(os.rmdir, fullname, sys.exc_info())
                        else:
                            try:
                                # This can only happen if someone replaces
                                # a directory with a symlink after the call to
                                # stat.S_ISDIR above.
                                raise OSError("Cannot call rmtree on a symbolic "
                                              "link")
                            except OSError:
                                onerror(os.path.islink, fullname, sys.exc_info())
                    finally:
                        os.close(dirfd)
            else:
                try:
>                   os.unlink(name, dir_fd=topfd)
E                   FileNotFoundError: [Errno 2] No such file or directory: 'libgkl_compression2251932993111688424.so'
tools/conda-tools/default/lib/python3.6/shutil.py:436: FileNotFoundError
```

--- a/tools/kraken.py
+++ b/tools/kraken.py
@@ -172,7 +172,7 @@ class Kraken(tools.Tool):
                 log.debug('Calling kraken output command line: %s', cmd)
                 subprocess.check_call(cmd, shell=True, executable='/bin/bash', env=env)
 
-                if bam2fq_ps.poll()
+                if bam2fq_ps.poll():
                     raise subprocess.CalledProcessError(bam2fq_ps.returncode, "SamToFastqTool().execute({})".format(in_bam))
 
 

--- a/tools/kraken.py
+++ b/tools/kraken.py
@@ -137,9 +137,9 @@ class Kraken(tools.Tool):
 
                 if outReads:
                     if outReports:
-                        cmd += ' | tee >(gzip --best > {kraken_reads})'
+                        cmd += ' | tee >(pigz --best > {kraken_reads})'
                     else:
-                        cmd += ' | gzip --best > {kraken_reads}'
+                        cmd += ' | pigz --best > {kraken_reads}'
 
                     cmd = cmd.format(kraken_reads=outReads[i])
 
@@ -165,12 +165,15 @@ class Kraken(tools.Tool):
                     'CLIPPING_ATTRIBUTE': tools.picard.SamToFastqTool.illumina_clipping_attribute,
                     'CLIPPING_ACTION': 'X'
                 }
-                picard.execute(in_bam, fastq_pipes[i*2], fastq_pipes[i*2 + 1],
-                               picardOptions=tools.picard.PicardTools.dict_to_picard_opts(picard_opts),
-                               JVMmemory=picard.jvmMemDefault, background=True)
+                bam2fq_ps = picard.execute(in_bam, fastq_pipes[i*2], fastq_pipes[i*2 + 1],
+                    picardOptions=tools.picard.PicardTools.dict_to_picard_opts(picard_opts),
+                    JVMmemory=picard.jvmMemDefault, background=True)
 
                 log.debug('Calling kraken output command line: %s', cmd)
                 subprocess.check_call(cmd, shell=True, executable='/bin/bash', env=env)
+
+                if bam2fq_ps.poll()
+                    raise subprocess.CalledProcessError(bam2fq_ps.returncode, "SamToFastqTool().execute({})".format(in_bam))
 
 
     def classify(self, inBam, db, outReads, numThreads=None):

